SwiftAA/Angles.swift

```diff
@@ -11,14 +11,34 @@ import Foundation
 /// The Degree is a unit of angle.
 /// Degree structs conform to SwiftAA Numeric type protocol.
 public struct Degree: NumericType, CustomStringConvertible {
-    /// The Degree value
-    public let value: Double
     
+    public var magnitude: Double
+    
+    public static func -=(lhs: inout Degree, rhs: Degree) {
+        lhs.value = lhs.value - rhs.value
+    }
+    
+    public static func +=(lhs: inout Degree, rhs: Degree) {
+        lhs.value = rhs.value + lhs.value
+    }
+    
+    public static func *(lhs: Degree, rhs: Degree) -> Degree {
+        return Degree(lhs.value * rhs.value)
+    }
```

Comment:
The problem with falling back to the generic `NumericType` is that we can't call `NumericType(newValue)` as it's a protocol, right? I just tried moving that method up into `NumericType` and I get this error: `Protocol 'NumericType' can only be used as a generic constraint because it has Self or associated type requirements`

This is the code
``` swift
 public static func +=(lhs: inout NumericType, rhs: NumericType) {
        lhs.value = rhs.value + lhs.value
    }
```

I'm in no huge hurry for this. I'm able to use this branch for my project so I feel as though we can use this as a learning experience. If there is a proper swifty way of doing things, I would strongly prefer it.